src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -607,7 +609,15 @@ private void UpdateTextBoxes(PackageSourceContextInfo packageSource)
             }
         }
 
-        private void OnBrowseButtonClicked(object sender, EventArgs e)
+        private void OnBrowseButtonClicked(object sender, EventArgs args)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
+            {
+                await OnBrowseButtonClickedAsync();
+            }).PostOnFailure(nameof(PackageSourcesOptionsControl), nameof(OnBrowseButtonClicked));
+        }
+
+        private async Task OnBrowseButtonClickedAsync()
         {
             ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
Given that this is now async, this check is incorrect. 

See rule #1 https://github.com/microsoft/vs-threading/blob/main/doc/threading_rules.md

Reply:
Ok. Replaced with `await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();`